Feedback1

durin93 opened the pull request:
개인프로젝트 코드리뷰 부탁드립니다.

구현 기능은 다음과 같습니다.

메모 기능
메모 추가, 수정 , 삭제
검색 기능
페이징 기능

링크 기능
링크 즐겨찾기 기능(추가, 수정, 삭제)
(링크 추가 시 해당 url의 title과 썸네일 이미지가 출력됨)

라벨 기능
라벨을 통한 메모, 링크 관리

친구 기능
친구 추가
친구 요청 수락 및 거절

회원 기능
일반 회원가입
oAuth2(kakao api) 회원가입

대부분의 기능은 RestController & Ajax로 구현하였습니다.

durin93 committed:
init

durin93 committed:
clean project

durin93 committed:
추가

javajigi commented on src/main/java/com/durin/UnAuthenticationException.java:
```diff
@@ -1,26 +1,26 @@
+
+    public UnAuthenticationException(Throwable cause) {
+        super(cause);
+    }
```
Exception 상속할 때 모든 생성자를 override해야 하나?

javajigi commented on src/main/java/com/durin/WebMvcConfig.java:
```diff
@@ -1,82 +1,82 @@
+	
+	@Bean
+	public Encrpytion encryption() {
+		return new Encrpytion(passwordEncoder());
```
굳이 Encryption으로 di를 받아야 되나?
PasswordEncoder가 필요한 곳에 직접 di하면 안되나?

javajigi commented on src/main/java/com/durin/domain/Labels.java:
```diff
@@ -1,34 +1,34 @@
+	private List<Label> labels = new ArrayList<>();
+
+	public List<Label> getLabels() {
+		return labels;
```
외부에서 변경할 수 없도록 Collections의 unmodifiableList인가 메소드 활용해 감싼 후 반환

javajigi commented on src/main/java/com/durin/domain/Labels.java:
```diff
@@ -1,34 +1,34 @@
+		return labels;
+	}
+	
+	public int AllCount() {
```
메소드 이름은 소문자

javajigi commented on src/main/java/com/durin/domain/Labels.java:
```diff
@@ -1,34 +1,34 @@
+		return labels.size();
+	}
+	
+	public int AllMemoCount() {
```
메소드 이름은 소문자

javajigi commented on src/main/java/com/durin/domain/Link.java:
```diff
@@ -1,83 +1,83 @@
+	@JsonIgnore
+	private User writer;
+	
+	private String title;
```
null 허용이 맞나?
Column 애노테이션 활용해 좀 더 철저하게 칼럼 설정한다.

javajigi commented on src/main/java/com/durin/domain/Link.java:
```diff
@@ -1,83 +1,83 @@
+		this.title = link.title.trim();
+		this.content = link.content;
+		this.originalUrl = link.originalUrl;
+		this.thumnailUrl = "https://image.thum.io/get/width/270/crop/800/"+link.originalUrl;
```
중복 제거. 상수 값으로 처리한다.

javajigi commented on src/main/java/com/durin/domain/MemoRepository.java:
```diff
@@ -1,32 +1,32 @@
+
+	List<Memo> findByLabelAndContentLikeOrTitleLike(Label label, String content, String title);
+
+	List<Memo> findByWriter(User loginUser);
```
이렇게 많은 find 절이 필요한 이유는?
재사용할 수 있는 방법은 없나?

javajigi reviewed (changes requested):
코드는 전체적으로 깔끔하게 구현했네. 아쉬움이라면 단위 테스트, acceptance test가 없는 것이 아쉽다. 테스트 코드 추가해 보면 좋겠다.

durin93 committed:
- Exception 필요 없는 생성자 제거
- BCrypt 필요시 DI
- Collections unmodifiableList 적용
- 메서드 이름 소문자....로 변경
- Entity Column 설정
- Entity 중복 제거, 상수 처리
- Repository Find 제거
- add domainTestCode
;
